Shared/ComponentParent/src/GizmoComponent.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { Fragment, useEffect, useRef, useState } from "react";
+import {
+    Action,
+    ActionManager,
+    BoundingBoxGizmo,
+    Color3,
+    ExecuteCodeAction,
+    Mesh,
+    PointerDragBehavior,
+    Tools,
+    Vector3
+} from "@babylonjs/core";
+
+export function GizmoComponent(props: {
+    mesh?: Mesh;
+    draggingEnabled: boolean;
+    pinchEnabled: boolean;
+    rotationEnabled: boolean;
+    gizmoSize: number;
+    onScale: (newScale: Vector3) => void;
+    onDrag: (newPosition: Vector3) => void;
+    onRotate: (newRotation: Vector3) => void;
+    color: string;
+}): React.ReactElement {
+    const [gizmo, setGizmo] = useState<BoundingBoxGizmo>();
+    const [dragBehaviour, setDragBehaviour] = useState<PointerDragBehavior>();
+    const gizmoRef = useRef<BoundingBoxGizmo>();
+    const [action, setAction] = useState<Action>();
+    const meshRef = useRef<Mesh>();
+
+    useEffect(() => {
+        setAction(new ExecuteCodeAction(ActionManager.OnPickUpTrigger, () => props.onDrag(meshRef.current!.position)));
+        return () => {
+            gizmoRef.current?.dispose();
```

Comment:
How I understand it, I need to clean up the gizmo when the component is closed. However, when I'm setting this, the gizmo element isn't set yet. So it'll be empty. If I set the gizmo in a ref, it will be able to find it.

Reply:
Oh wait, I think I understand what you mean. Let me check

Reply:
Yeah you're right, I've changed it to a more logical implementation